src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -204,40 +202,18 @@ export const Deposit: React.FC<DepositProps> = ({
     enableEstimateFees: Boolean(!isApprovalRequired && bnOrZero(inputValues?.cryptoAmount).gt(0)),
   })
 
-  useEffect(() => {
-    if (!accountId) return
-    ;(async () => {
-      const isApprovalRequired = await (async () => {
-        // Router contract address is only set in case we're depositting a token, not a native asset
-        if (!inboundAddress) return false
-        // Do not try to get allowance for native assets, including non-EVM ones.
-        if (!isTokenDeposit) return false
-
-        const allowanceOnChainCryptoBaseUnit = await getErc20Allowance({
-          address: fromAssetId(assetId).assetReference,
-          spender: inboundAddress,
-          from: fromAccountId(accountId).account,
-          chainId: asset.chainId,
-        })
-
-        const cryptoAmountBaseUnit = toBaseUnit(inputValues?.cryptoAmount, asset.precision)
-
-        if (bnOrZero(allowanceOnChainCryptoBaseUnit).eq(0)) return true
-        if (bn(cryptoAmountBaseUnit).gt(allowanceOnChainCryptoBaseUnit)) return true
+  const { isAllowanceResetRequired, isApprovalRequired: _isApprovalRequired } =
+    useIsApprovalRequired({
+      assetId,
+      // TODO(gomes): consolidate
+      amountCryptoBaseUnit: toBaseUnit(inputValues?.cryptoAmount, asset?.precision ?? 0),
+      spender: inboundAddress,
+      from: accountId ? fromAccountId(accountId).account : undefined,
+    })
 
-        return false
-      })()
-      setIsApprovalRequired(isApprovalRequired)
-    })()
-  }, [
-    accountId,
-    asset.chainId,
-    asset.precision,
-    assetId,
-    inputValues,
-    inboundAddress,
-    isTokenDeposit,
-  ])
+  useEffect(() => {
+    setIsApprovalRequired(_isApprovalRequired)
+  }, [_isApprovalRequired, setIsApprovalRequired])
```

Comment:
preferable-blocking: looks super useless, could you explain the issue you found requiring to implement this effect?

Reply:
chicken and egg scenario, same like the one below - we need `isApprovalRequired` to disable fees estimation in `useSendThorTx()`, but we need the inbound address as spender for `useIsApprovalRequired()`